src/server/pfs/server/driver_fsck.go

```diff
@@ -338,10 +339,7 @@ func (d *driver) fsck(ctx context.Context, fix bool, cb func(*pfs.FsckResponse)
 					}
 					contains := false
 					for _, headProv := range headCommitInfo.Provenance {
-						if provBranchInfo.Head.Branch.Repo.Name == headProv.Commit.Branch.Repo.Name &&
-							provBranchInfo.Branch.Repo.Name == headProv.Commit.Branch.Repo.Name &&
-							provBranchInfo.Branch.Name == headProv.Commit.Branch.Name &&
-							provBranchInfo.Head.ID == headProv.Commit.ID {
+						if proto.Equal(provBranchInfo.Head, headProv.Commit) {
```

Comment:
I wasn't sure if we tend to avoid `proto.Equal`, I can definitely understand not wanting to use `reflect` more than necessary